chapters/traversal.md

```diff
@@ -86,83 +88,67 @@ from jina import Document
-Each shot includes one or more frames. Such hierarchical structures can be very well represented with the recursive representation.
+This recursive structure provides Jina the flexibility to cover any complex use case that may require search at different semantic units. Besides that, the recursive structure enables Jina rankers to accumulate scores from lower granularities to upper granularities, such as `Chunk2DocRankers`. For example, in NLP a long document is composed of semantic chapters; each chapter consists of multiple paragraphs, which can be further segmented into sentences. In CV, a video is composed of one or more scenes, including one or more shots (i.e. a sequence of frames taken by a single camera over a continuous period of time). Each shot includes one or more frames. 
 
-If we look from a tree view (with a depth of 3):
+Such hierarchical structures can be very well represented with the recursive representation. The image below shows the tree view (with a depth of 3):
 
 ![tree-view](./images/tree.png)
 
-## Document Traversal with traversal paths
+## Document traversal with traversal paths
 
-As you already learned from [Jina 101](https://101.jina.ai), you needs to apply transformation (i.e. a `callback`) on a different level of documents.
-Given the tree structure, how could we achieve that?
-The answer is `traversal`.
+As you have already learned from [Jina 101](https://101.jina.ai), you need to apply transformation (i.e. a `callback`) on a different level of documents. Given the tree structure, how could we achieve that? The answer is `traversal`.
 
-Jina has defined a method called `traversal` within the class of Document, which looks like this:
+Jina has defined a method called `traversal()` within the class of a Document. The method looks like this:
 
 ```python
 def traverse(self, traversal_path: str, callback_fn: Callable, *args, **kwargs) -> None
     """Traversal apply `:meth:callback_fn` on the recursive tree representation."""
     ...
 ```
 
-This allows you to apply `callback_fn` based on `traversal_path`.
-The `traversal_path` is defined as below:
+This allows you to apply a callback function named `callback_fn()` based on `traversal_path`. The `traversal_path` is defined as shown below:
 
 ![nodes](./images/nodes.png)
 
-With these pre-defined node names, you're able to apply any callbacks (defined as `_apply_all` in the `driver`) to a specific node.
-In the below YAML configuration, the `VectorSearchDriver` was applied to node `c`, `KVSearchDriver` was applied to node `cm` (matches of chunks).
+With these pre-defined node names, you are able to apply any callbacks (defined as `_apply_all` in the `driver`) to a specific node. In the YAML configuration below, the `VectorSearchDriver` was applied to node `c`, `KVSearchDriver` was applied to node `cm` (matches of chunks).
```

Comment:
It would be good that we add some explanations why VectorSearchDriver is 'c' and KVSearchDriver is 'cm'. 
My understanding is: The 2 drivers will be applied sequentially, so we first apply VectorSearchDriver for the indexer. It will help to get matches for chunks(cm) and add them, the matches for chunks will contain scores and doc.id. Then we will apply KVSearchDriver at matches of chunks(cm) to get the embeddings for 'cm' via the doc.id.

Reply:
Your right @Yongxuanzhang, but I don't think Frank has this knowledge.. we'll need to find someone else in the team.

Reply:
Well, @FionnD is right. I just accepted the code as is, and did not think about it too much. I also like your explanation @Yongxuanzhang , and I think of adding it below the code.